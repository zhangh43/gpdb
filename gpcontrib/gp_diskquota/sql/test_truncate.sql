-- Test truncate
CREATE SCHEMA s7;
SELECT diskquota.set_schema_quota('s7', '1 MB');
SET search_path TO s7;
CREATE TABLE a (i int);
CREATE TABLE b (i int);
INSERT INTO a SELECT generate_series(1,100000000);
SELECT pg_sleep(5);
-- expect insert fail
INSERT INTO a SELECT generate_series(1,30);
INSERT INTO b SELECT generate_series(1,30);
TRUNCATE TABLE a;
SELECT pg_sleep(5);
-- expect insert succeed
INSERT INTO a SELECT generate_series(1,30);
INSERT INTO b SELECT generate_series(1,30);

DROP TABLE a, b;
RESET search_path;
DROP SCHEMA s7;

