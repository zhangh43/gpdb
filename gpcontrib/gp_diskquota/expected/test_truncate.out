-- Test truncate
create schema s7;
select diskquota.set_schema_quota('s7', '1 MB');
 set_schema_quota 
------------------
 
(1 row)

set search_path to s7;
create table a (i int);
create table b (i int);
insert into a select generate_series(1,50000);
select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- expect insert fail
insert into a select generate_series(1,30);
ERROR:  schema's disk space quota exceeded with name:s7
insert into b select generate_series(1,30);
ERROR:  schema's disk space quota exceeded with name:s7
truncate table a;
select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- expect insert succeed
insert into a select generate_series(1,30);
insert into b select generate_series(1,30);
drop table a, b;
reset search_path;
drop schema s7;
